orbit/cmd/orbit/orbit.go

```diff
@@ -515,6 +510,90 @@ func main() {
 		}
 		g.Add(flagRunner.Execute, flagRunner.Interrupt)
 
+		trw := token.NewReadWriter(filepath.Join(c.String("root-dir"), "identifier"))
+
+		if err := trw.LoadOrGenerate(); err != nil {
+			return fmt.Errorf("initializing token read writer: %w", err)
+		}
+
+		if orbitClient.GetServerCapabilities().Has(fleet.CapabilityOrbitEndpoints) &&
```

Comment:
Guessing we don't need to ping because the flags updater above uses the `orbitClient` to do an initial flags fetch (therefore populating headers)?

Reply:
that's right, on second thought it might be a good idea to add an explicit check + comment here for clarity

Reply:
If this invalidates testing I don't think we should add the ping :) (am ok with current changes, maybe just a comment)

Reply:
what I'm thinking is: what if we move the flag check somewhere else? we don't have any automated tests that will catch that, so I wonder if it's better to be explicit.

Reply:
> what if we move the flag check somewhere else?

Has to be called early at start up to change any flags before starting osquery.

Reply:
gotcha, I meant: there's an explicit requirement for the calls to execute in a particular order, which might break things without us knowing, but I can't argue with reducing the server load 🗡️ 

I replaced the ping call with a comment